File: Java/ToyLanguageGUI_exam_final/src/domain/IFileTable.java
package domain;

import java.util.Collection;
import java.util.HashMap;

public interface IFileTable<K,V> {
    V get(K key);
    boolean contains(K key);
    void remove(K key);
    void add(K key,V value);
    Iterable<K> getAll();
    Iterable<V> getValues();
    HashMap<K,V> getContent();
}

File: ToyInterpretor/ToyInterpretor/Domain/IFileTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ToyInterpretor.Domain
{
    interface IFileTable<K,V>
    {
        V get(K key);
        Boolean contains(K key);
        void remove(K key);
        void add(K key, V value);
        IEnumerable<K> getAll();
        IEnumerable<V> getValues();
    }
}
